libpromises/files_names.c

```diff
@@ -568,8 +572,7 @@ int CompressPath(char *dest, const char *src)
 
         AddSlash(dest);
 
-        /* TODO use dest_size parameter instead of CF_BUFSIZE. */
-        size_t ret = strlcat(dest, node, CF_BUFSIZE);
+        size_t ret = strlcat(dest, node, dest_length);
 
         if (ret >= CF_BUFSIZE)
         {
```

Comment:
There is a "memset(dest, 0, dest_length);" at the top of the function, so I didn't change this. It's a bit unclear to me if you mean something else? In that case I would handle it separately (not related to this change?).